doc/cable_diag/cable_diag.md

```diff
@@ -0,0 +1,279 @@
+<img src="./images/ocp.png"
+style="width:5.0in;height:6.2in" />
+
+<table>
+<colgroup>
+<col style="width: 100%" />
+</colgroup>
+<tbody>
+</tbody>
+</table>
+
+#Switch Abstraction Interface Change Proposal
+
+<table>
+<colgroup>
+<col style="width: 16%" />
+<col style="width: 83%" />
+</colgroup>
+<thead>
+<tr class="header">
+<th><strong>Title</strong></th>
+<th><strong>Cable Diagnostics Proposal</strong></th>
+</tr>
+</thead>
+<tbody>
+<tr class="odd">
+<td><strong>Authors</strong></td>
+<td><strong>Broadcom Inc</strong></td>
+</tr>
+<tr class="even">
+<td><strong>Status</strong></td>
+<td><strong>In Review</strong></td>
+</tr>
+<tr class="odd">
+<td><strong>Type</strong></td>
+<td><strong>Standards Track</strong></td>
+</tr>
+<tr class="even">
+<td><strong>Created</strong></td>
+<td><strong>10/06/2023</strong></td>
+</tr>
+<tr class="odd">
+<td><strong>SAI-Version</strong></td>
+<td><strong>V0.0</strong></td>
+</tr>
+</tbody>
+</table>
+
+- [1 Overview](#Overview)
+
+- [2 Cable diagnostics Use Case](#UseCase)
+
+- [3 Attribute Specification](#AttrSpec)
+
+- [3.1 Changes in file saiport.h](#Changes)
+
+- [4 Examples](#Examples)
+
+- [4.1 Cable Diagnostics Usage](#Usage)
+
+<a id="Overview"></a>
+## 1 Overview
+
+When a link is up and operating normally with adequate bit error
+performance, there is no need for any diagnostic either on the
+digital/analog portion of the link or on the cable connecting the link.
+However, the characteristics of the problems associated with a cable
+become of paramount concern when a link cannot be established or is
+performing poorly.
+
+Enhanced Cable Diagnostics (ECD) provides with intelligent
```

Comment:
@prafull-brcm where is ECD implemented on the hardware? On the ASIC? Serdes core?

Reply:
@prgeor  ECD is implemented inside the PHY chip HW.